client/src/components/pages/EditProfile.js

```diff
@@ -1,81 +1,164 @@
-import React, { useEffect, useState } from "react";
+import React, { useState, useEffect } from "react";
 import { Container, Form, Button, Row, Col } from "react-bootstrap";
-import { useAuth } from "../../context/auth/AuthState";
+import { deleteUser, updateUser, useAuth, clearErrors } from "../../context/auth/AuthState";
+import AlertEditProfile from '../Alerts/AlertEditProfile';
+import AlertModal from "../Alerts/AlertModal";
+import { LOGOUT, UPDATE_USER_SUCCESS, UPDATE_USER_FAIL } from "../../context/types";
 
 function EditProfile() {
-	const [authState] = useAuth();
-	const { isAuthenticated, user } = authState;
-	const [validated, setValidated] = useState(false);
+	const [authState, authDispatch] = useAuth();
+	const { isAuthenticated, user, error} = authState;
+	const [alert, setShowAlert] = useState(false);
 
+	const [userData, setUserData] = useState({
+		firstName: '',
+		lastName: '',
+		organization: '',
+		email: ''
+	});
 
-	const handleSubmit = (event) => {
-		const form = event.currentTarget;
-		if (form.checkValidity() === false) {
-			event.preventDefault();
-			event.stopPropagation();
+	useEffect(() => {
+		setUserData({
+			firstName: isAuthenticated && user && user.firstName,
+			lastName: isAuthenticated && user && user.lastName,
+			organization: isAuthenticated && user && user.organization,
+			email: isAuthenticated && user && user.email,
+		});
+	}, [isAuthenticated, user, authDispatch]);
+
+
+	useEffect(() => {
+		if (UPDATE_USER_FAIL) {
+			console.log(UPDATE_USER_FAIL.payload);
+			setShowAlert(true);
+			
+		}
+	}, [error, isAuthenticated, authDispatch]);
+	// const findErrors(() => {
+	// 	const alphaReg = /^[a-zA-Z]+$/;
+	// 	const { firstName, lastName, email, organization } = form
+	// 	const newErrors = {}
+	// 	if (!firstName || firstName === '') newErrors.firstName = 'First name cannot be blank.'
+	// 	else if (firstName.length < 1 || firstName.length > 30) newErrors.firstName = 'First name must be between 1 and 30 characters long.'
+	// 	else if (!alphaReg.test(firstName)) newErrors.firstName
+	// });
+
+	const {
+		firstName,
+		lastName,
+		organization,
+		email,
+	} = userData;
+
+
+	const onChange = (event) => {
+		setUserData({ ...userData, [event.target.name]: event.target.value });
+		console.log(userData)
+		setShowAlert(false);
+		clearErrors(authDispatch);
+	};
+	
+	const onSubmit = (event) => {
+
+		event.preventDefault();
+
+		updateUser(authDispatch, userData, user._id);
```

Comment:
it seems like onchange is console logging correctly, and postman functionalities are working fine, I assume there's something in between that's causing the problem and I'm not sure why

Reply:
It might that you are setting the user info in useEffect thus it was always be the state it was originally set to when the page re-renders. I'm not entirely sure but I will see what the issue could be